### PR TITLE
Navbar - new ui + right menu

### PR DESCRIPTION
**Pull request by wintersummermint**
### All Submissions

Navbar is changed and links are placed , this approach is iterational progress for eventually new ui revamp.

Note : 
withdraw button is not enabled, they can claim from old way.
Escape button key to close modal not available yet on this PR.
Mobile enhanced support is on the next PR upcoming
payout chain dropdowns is on next upcoming PR's

* [ ] Can you post a screenshot of your changes (if applicable)?
jUdBX1Axh2

### New Feature Submissions

* [ ] Does this relate to an existing issue, or has it been talked about prior to being done (if a major change)?

### Notes

Please note, your code must pass all tests and lint checks before it can be merged.

### PR Description

(insert description here)

### Testing

Has the code been tested, or does it need double checking?

### CODE REVIEW

**Comment by seiyria** [comment]
please _only_ commit files that are relevant to this PR. I have no idea what most of these assets are, or why we need them all right now. if they're needed later, they can be added later.

also, the names of these files are jank. there is no reason for (2), or a random string of numbers should precede the file name. please rename them to be accurate.

additionally, some of these assets are uppercase starting, while others are lowercase. I believe the project convention is lowercase.

lastly, when adding to a locale file, you need to add the same text to all locale files, even if the text is untranslated, or else it creates bad-looking artifacts.

**Review by seiyria — DISMISSED** [review]
lots of extraneous assets, asset names need to be addressed, etc (see other comment)

**Review by Chriserus — CHANGES_REQUESTED** [review]
Unmergable

**Comment by Chriserus** [comment]
Sei is correct with almost all the remarks, you don't need to (even _shouldn't_) add untranslated text to other locales, as it defaults to English (we have a fallback), no weird artifacts @seiyria 
Also, for sure Avax isn't Water in English
image

Btw where is the admin panel? I thought you're working on some old branch when it's not there yet, but nope, you deleted it 
image

**Comment by wintersummermint** [comment]
> Btw where is the admin panel? I thought you're working on some old branch when it's not there yet, but nope, you deleted it image

Fixing this, i must been overlooked after redoing the ui it doesnt exist in adobeXD i will add
image

**Comment by Chriserus** [comment]
General comments:
- the navbar doesn't look correct on my end
image
- Assets. You're still using the new-ui folder, don't, just place them in appropriate places in assets/subfolders. Also, some of them still have meaningless names (example: [6351903_burger_list_menu_navigation_icon@2x.png])
- also, the same asset of skill token got committed twice
- If you didn't add/delete any libraries, don't commit package-lock.json
- Closing it with "ESC" button didn't work for me
image
- Tooltip is totally off
image
- I would consider adding some links that are often viewed (Quests) to the main bar, in current state I have to click on the burger menu and from there on quests. I understand why Stake can be there, or utilities, but Quests, Raids, Combat, Blacksmith, PvP etc. should be on the main navbar
- Why does "Withdraw" seems clickable, but it does nothing on click?
- When I changed Bazaar to be an "external" link (opening it in the new tab), I also introduced an icon that represents it, I think it's a good idea to somehow show to the user that this particular item will be opened in a separate tab (same for Wiki for example)
- Main "CryptoBlades" is clickable, but does nothing, either delete the click indication, or link it to plaza

File: frontend/src/components/Options.vue [hunk 1]
```diff
@@ -1,47 +1,143 @@
...
+  <div class="body main-font options">
+    <img src="../assets/new-ui/6351903_burger_list_menu_navigation_icon@2x.png" class="burger-icon" @click="openMenu"/>
+    <div class="menu-open" v-if="isMenuOpen">
+      <div @click="closeMenu" class="x-button"><img src="../assets/new-ui/Close.png" /> Close (ESC)</div>
+      <div></div>
+      <div class="ads"></div>
+      <div class="menu-icons">
+        <div class="row-icons">
+          <router-link class="menu-icon" v-if="!stakeOnly" :to="{ name: 'plaza'}" exact>
+            <img src="../assets/new-ui/new-plaza-icon.png">
```
Comment (Chriserus) [inline frontend/src/components/Options.vue:11]: Also, for naming convention, don't use "new" on everything that's... new

File: frontend/src/components/Options.vue [hunk 2]
```diff
@@ -1,47 +1,143 @@
...
+            <img src="../assets/new-ui/new-blacksmith-icon.png">
+            <p>{{ $t("viewLink.blacksmith") }}</p>
+          </router-link>
+
+          <router-link class="menu-icon" v-if="!stakeOnly" :to="{ name: 'combat'}" exact>
+            <img src="../assets/new-ui/new-combat-icon.png">
+            <p>{{ $t("viewLink.combat") }}</p>
+          </router-link >
+
+          <router-link class="menu-icon" v-if="pvp" :to="{ name: 'pvp'}" exact>
```
Comment (Chriserus) [inline frontend/src/components/Options.vue:25]: Why is arena link here twice? If PvP is not turned on, you can just change the class conditionally. Also, where is the functionality for checking if particular chain supports Quests/PvP/Merch (and informing user about it)?

Comment (Chriserus) [reply frontend/src/components/Options.vue:25]: Same for the things below, you can just switch the css class for turned off functionalities

File: frontend/src/components/Options.vue [hunk 3]
```diff
@@ -1,47 +1,143 @@
...
+        </div>
 
-        <b-dropdown-item :to="{ name: 'bridge' }" v-if="isBridgeEnabled">
-           <i class="fa fa-exchange-alt mr-2"></i>{{$t('optionsMenu.bridgeNfts')}}
-        </b-dropdown-item>
+         <div class="row-icons">
+            <router-link class="menu-icon" :to="{ name: 'options'}" exact>
+            <img src="../assets/new-ui/new-gear-icon.png">
+            <p>{{ $t("viewLink.settings") }}</p>
+            </router-link>
```
Comment (Chriserus) [inline frontend/src/components/Options.vue:105]: Formatting is off (indentations)

File: frontend/src/components/Options.vue [hunk 4]
```diff
@@ -137,18 +238,42 @@ export default Vue.extend({
...
-      ClaimStage
+      ClaimStage,
+      stakeOnly,
+      raid,
+      market,
+      portal,
+      pvp,
+      quests,
+      merchandise,
+      hasAdminAccess: false,
```
Comment (Chriserus) [inline frontend/src/components/Options.vue:254]: I don't think you're setting this value anywhere else

File: frontend/src/components/Options.vue [hunk 5]
```diff
@@ -172,6 +296,22 @@ export default Vue.extend({
...
+      const supportedProjects = this.getPartnerProjects.map(p => {
+        return {
+          id: p.id,
+          name: p.name,
+          tokenSymbol: p.tokenSymbol,
+          tokenAddress: p.tokenAddress,
+          tokenSupply: p.tokenSupply,
+          tokensClaimed: p.tokensClaimed,
+          tokenPrice: p.tokenPrice,
+          isActive: p.isActive
```
Comment (Chriserus) [inline frontend/src/components/Options.vue:309]: What is this mapping for? Am I missing something or are you mapping all the fields to the same fields?

File: frontend/src/components/Options.vue [hunk 6]
```diff
@@ -239,4 +387,102 @@ export default Vue.extend({
...
+}
+.burger-icon {
+  width: 35px;
+}
+.burger-icon:hover {
+  cursor: pointer;
+}
+
+.menu-open {
+    height: 100%;
```
Comment (Chriserus) [inline frontend/src/components/Options.vue:405]: Formatting (spacing + indentations)

File: frontend/src/components/Options.vue [hunk 7]
```diff
@@ -239,4 +387,102 @@ export default Vue.extend({
...
+.menu-icon {
+  width: 95px;
+  height: 95px;
+  background: #1A253B 0% 0% no-repeat padding-box;
+  border: 1px solid #344362;
+  border-radius: 5px;
+  opacity: 1;
+  text-align: center;
+  padding-top:20px;
+  color : white;
```
Comment (Chriserus) [inline frontend/src/components/Options.vue:447]: When you're copying css from the designs, try to trim unnecessary styles (like opacity: 1 for example)

File: frontend/src/components/P2EButton.vue [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+
+<script lang="ts">
+
+export default {
+  props: ['mainText', 'subText', 'disabled', 'route', 'size', 'targetRef'],
+};
+
+</script>
+
+<style scoped>
```
Comment (Chriserus) [inline frontend/src/components/P2EButton.vue:24]: A lot of styles for such component, do we need all of them? Also, where is this component used?

File: frontend/src/components/ViewLinks.vue [hunk 9]
```diff
@@ -1,77 +1,51 @@
...
+      <router-link v-if="!stakeOnly" :to="{ name: 'plaza' }" exact class="nav-link">
+        <div class="icon"> <img src="../assets/new-ui/new-plaza-icon.png" class="new-ui-links-icon"></div>
+        <div class="link-text">Character</div>
+      </router-link>
+    </li>
+
+   <li v-if="!stakeOnly" >
+      <router-link :to="{ name: 'blacksmith' }" exact class="nav-link">
+        <div class="icon"><img src="../assets/new-ui/new-blacksmith-icon.png" class="new-ui-links-icon"></div>
+        <div class="link-text">Blacksmith</div>
```
Comment (Chriserus) [inline frontend/src/components/ViewLinks.vue:17]: Missing translations in places that already had translations... Look at the deleted code

File: frontend/src/components/ViewLinks.vue [hunk 10]
```diff
@@ -1,77 +1,51 @@
 <template>
   <b-navbar-nav>
+    <!-- <li class="play-to-earn top-nav-links">
+      <P2EButton mainText="Play-to-earn" route="play-to-earn" />
+    </li> -->
+
+    <li class="character top-nav-links" >
```
Comment (Chriserus) [inline frontend/src/components/ViewLinks.vue:7]: That's why the styles were "needed" there app-scoped! I think you can just use a div, not a list element

File: frontend/src/components/ViewLinks.vue [hunk 11]
```diff
@@ -1,77 +1,51 @@
...
+
+    <li v-if="!stakeOnly" >
+      <router-link :to="{ name: 'combat' }" exact class="nav-link">
+        <div class="icon"><img src="../assets/new-ui/new-combat-icon.png" class="new-ui-links-icon"></div>
+        <div class="link-text">Combat</div>
+      </router-link>
+    </li>
+
+    <li v-if="pvp" >
+      <router-link :to="{ name: 'pvp' }" exact class="nav-link">
```
Comment (Chriserus) [inline frontend/src/components/ViewLinks.vue:29]: No "disabled" for unsupported chain. Missing tooltip from before

File: frontend/src/components/ViewLinks.vue [hunk 12]
```diff
@@ -80,7 +54,8 @@
 import {market, merchandise, portal, pvp, quests, raid, stakeOnly} from '@/feature-flags';
 import {mapActions, mapGetters, mapState} from 'vuex';
 import Vue from 'vue';
-import Hint from '@/components/Hint';
+// import Hint from '@/components/Hint';
+// import P2EButton from '@/components/P2EButton';
```
Comment (Chriserus) [inline frontend/src/components/ViewLinks.vue:58]: Are the comments here intentional?

File: frontend/src/components/ViewLinks.vue [hunk 13]
```diff
@@ -119,7 +94,8 @@ export default Vue.extend({
   },
 
   components: {
-    Hint,
+    // Hint,
+    // P2EButton
```
Comment (Chriserus) [inline frontend/src/components/ViewLinks.vue:98]: Same here

File: frontend/src/components/smart/SkillBalanceDisplay.vue [hunk 14]
```diff
@@ -14,17 +14,25 @@
...
-        @mouseover="hover = !isMobile() || true"
-        @mouseleave="hover = !isMobile()"
-      >{{ formattedTotalSkillBalance }} <b-icon-gift-fill scale="1" v-if="hasInGameSkill" variant="success"/>
-      </span>
+        <p>$SKILL </p>
+        <b v-tooltip="{ content: totalSkillTooltipHtml , trigger: (isMobile() ? 'click' : 'hover') }"
+          @mouseover="hover = !isMobile() || true"
+          @mouseleave="hover = !isMobile()"
+        >{{ formattedTotalSkillBalance }} <b-icon-gift-fill scale="1" v-if="hasInGameSkill" variant="success"/></b>
+        <!-- <span class="balance"
```
Comment (Chriserus) [inline frontend/src/components/smart/SkillBalanceDisplay.vue:26]: If leaving it commented out is necessary, please add a comment explaining why

**Review by Chriserus — CHANGES_REQUESTED** [review]

File: frontend/src/App.vue [hunk 15]
```diff
@@ -415,6 +415,9 @@ export default {
 </script>
 
 <style>
+
+ul, li {list-style-type: none;}
```
[the remark this answers is not in the document]
Comment (wintersummermint) [reply frontend/src/App.vue:419]: image
yes i am actually intending to make this universal , to make ul li unstyled as default not usual to have bulleted styling? , it appears to be happening on the screenshot as well

[the remark this answers is not in the document]
Comment (seiyria) [reply frontend/src/App.vue:419]: we don't want to make everything global; they should be disabled on an as-needed basis.

File: frontend/src/components/ViewLinks.vue (hunk 10 above)
Comment (wintersummermint) [reply frontend/src/components/ViewLinks.vue:7]: image
they we're part of navbar ul list , tried renaming them to div but make sense still to be list because they are part of navbar list ul li

**Comment by Chriserus** [comment]
Still looks incorrect on my side, but regardless:
image
- The check and tooltip for functionality supported is there for PvP, but not for Quests. Did you test it on your local environment? Also same for Merchandise (for now Merch is supported, so we shouldn't omit it). A tip for you to "mock" that the chain is not supprted:
image
For ui developing purposes that's enough to test it^
- I still do not see admin panel link, even though before I switched branches I saw it on the old ui, in the new one it doesn't show up. Did you actually run it and saw the admin panel on your local development environment?
- At the breakpoint seen on the screenshot, font changes to _really_ small and "deposit" is overlaying skill display 
image
- I think you missed my comment regarding Bazaar link to have an icon indicating that it will open in new tab
- The tooltip that I said was off is a little _less_ off now
image
The cause for that might be the "70px" size of the component here
image
After removing 70 px it's centered:
image
- Styles overriding in this place: what's with the !important and still having style: display: none
image

File: frontend/src/components/smart/SkillBalanceDisplay.vue [hunk 16]
```diff
@@ -200,7 +208,17 @@ export default Vue.extend({
...
 }
 .deposit-withdraw {
-  margin-top: 25px;
+  flex: 100%;
+  margin-top: -25px;
+  color: rgba(255, 255, 255, 0.364);
+
+  /* margin-top: 25px;
+  position: relative;
+  right: 35px; */
```
Comment (Chriserus) [inline frontend/src/components/smart/SkillBalanceDisplay.vue:217]: Commented out styles

**Review by Chriserus — DISMISSED** [review]

File: .vscode/tasks.json [hunk 17]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (Chriserus) [inline .vscode/tasks.json:1]: Why is this file committed? I think we have .vscode in gitignore

**Comment by wintersummermint** [comment]
image
Admin should be working

Cannot seem to replicate the condensed text on navbar 
tL7K693l7x
 
image
updated some fixes

File: frontend/src/App.vue [hunk 18]
```diff
@@ -61,13 +61,17 @@
...
         <small-button class="button mm-button" @click="toggleHideWalletWarning" :text="$t('app.warning.buttons.hide')" />
       </div>
-      <div class="ad-container">
-        <Adsense v-if="showAds && !isMobile()"
-          data-ad-client="ca-pub-6717992096530538"
-          data-ad-slot="5115599573"
-          data-ad-format="auto"
-          data-full-width-responsive="yes"
-          />
+      <div v-if="showAds && !isMobile()" class="ad-container">
```
Comment (Chriserus) [inline frontend/src/App.vue:64]: Why are these changes here? I think it was done in a separate PR

Comment (Chriserus) [reply frontend/src/App.vue:64]: Also same for CharacterBar and CombatResult

Comment (Chriserus) [reply frontend/src/App.vue:64]: And plaza. Is this branch up to date?

File: frontend/src/views/Quests.vue [hunk 19]
```diff
@@ -348,11 +348,15 @@ export default Vue.extend({
       if (this.nextWeekResetCheckInterval) {
         clearInterval(this.nextWeekResetCheckInterval);
       }
-      this.nextWeekResetCheckInterval = setInterval(() => {
+      this.nextWeekResetCheckInterval = setInterval(async () => {
```
Comment (Chriserus) [inline frontend/src/views/Quests.vue:351]: And this, why is this here?

**Comment by Chriserus** [comment]
So... as always, I double checked your changes for the admin panel view. I have main branch with local dev env spin up, I see admin panel link and I can go there. I switch to your branch, admin panel gone.
image

This time you posted a screenshot confirming that the admin panel actually _is_ there, did you have dev env running and actually checked that it works? Or did you mock it somehow?

In order to display admin panel: compile, deploy contracts, as you normally do for developing stuff. By default, first account is the one deploying the contracts and the one that has access to admin panel (as it has role assigned on deployment), and it's also the only one with skill by default. When you open the app and connect with this account, Admin panel should display (you can do a sanity check by switching to main branch, checking it's there, then switching to your own branch that _possibly_ has a problem).

image
For window width equal or less then 1316px (as seen on the screenshot attached) you won't be able to navigate any menu that's out of screen (until breakpoint, which in this case you set for 576px).

Also, I don't know what is the exact reason why I have a problem with text that overlays text from other links, but on the gif you provided 
image
We can see that SKILL display overlays itself even in your case, not only mine.

There are also some unrelated commits from different branches (my quests fix and coinzilla ads), why is that? How did you rebase/merge with main? Maybe something went wrong here, let me know

**Review by Sellie96 — APPROVED** [review]
